ofrak_core/src/ofrak/core/android_sparse.py

```diff
@@ -0,0 +1,195 @@
+import asyncio
+import logging
+import tempfile312 as tempfile
+from subprocess import CalledProcessError
+
+from ofrak.component.packer import Packer
+from ofrak.component.unpacker import Unpacker
+from ofrak.core.binary import GenericBinary
+from ofrak.core.magic import MagicDescriptionPattern, RawMagicPattern
+from ofrak.model.component_model import ComponentExternalTool
+from ofrak.resource import Resource
+from ofrak_type.range import Range
+
+LOGGER = logging.getLogger(__name__)
+
+# Android sparse image magic bytes: 0xed26ff3a in little-endian
+SPARSE_HEADER_MAGIC = b"\x3a\xff\x26\xed"
+
+
+class _AndroidSparseImageTool(ComponentExternalTool):
+    """
+    Custom tool checker for Android sparse image tools.
+
+    These tools don't support standard help flags and always return non-zero exit codes,
+    so we check for the presence of usage output instead.
+    """
+
+    def __init__(self, tool: str, apt_package: str, brew_package: str):
+        super().__init__(
+            tool, "https://github.com/anestisb/android-simg2img", "", apt_package, brew_package
+        )
+
+    async def is_tool_installed(self) -> bool:
+        """
+        Check if the tool is installed by running it with no arguments and checking for usage output.
+
+        :return: True if tool outputs usage information, False otherwise
+        """
+        try:
+            cmd = [self.tool]
+            proc = await asyncio.create_subprocess_exec(
+                *cmd,
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+            )
+            stdout, stderr = await proc.communicate()
+        except FileNotFoundError:
+            return False
+
+        # Check if usage information is present in output (tool returns 255, but that's OK)
+        output = stdout + stderr
+        if b"Usage:" in output or b"usage:" in output:
+            return True
+
+        return False
```

Comment:
Do we expect usage in stderr? If not, will the following not work?

```suggestion
            stdout, _ = await proc.communicate()
        except FileNotFoundError:
            return False

        # Check if usage information is present in output
        return b"usage:" in output.lower()
```